### PR TITLE
net: Enable ICMP by default if IPv4 is enabled

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Enable ICMP by default if IPv4 is enabled
## Impact
Avoid users missed basic configuration
## Testing
N/A

### CODE REVIEW

File: net/icmp/Kconfig [hunk 1]
```diff
@@ -7,7 +7,7 @@ menu "ICMP Networking Support"
 
 config NET_ICMP
 	bool "Enable ICMP networking"
-	default n
+	default NET_IPv4
```
Comment (xiaoxiang781216) [inline net/icmp/Kconfig:10]: but it isn't nessary to enable icmp when ipv4 is enabled

**Comment by acassis** [comment]
It not, but normally users will want to use it, so I suggest to keep simplify users life, please see Greg suggestion in the mailing list.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> It is not, but normally users will want to use it, so I suggest to keep it to simplify users life, please see Greg suggestion in the mailing list.

should we enable icmpv6 if ipv6 enable like ipv4? @acassis

**Comment by acassis** [comment]
> > It is not, but normally users will want to use it, so I suggest to keep it to simplify users life, please see Greg suggestion in the mailing list.
> 
> should we enable icmpv6 if ipv6 enable like ipv4? @acassis

Yes, I will submit a new PR

**Comment by simbit18** [comment]
@acassis  Linux (xtensa) ci fail, is it relative to this change?
https://github.com/apache/nuttx/actions/runs/9868544352/job/27250805029?pr=12658#step:7:1718

https://github.com/apache/nuttx/blob/738a30a4213a38158d3f8136faa6f563a054e424/boards/xtensa/esp32s3/esp32s3-devkit/configs/qemu_debug/defconfig#L61

**Pull request merged**